test-app/src/main/java/org/readium/r2/testapp/bookshelf/BookshelfFragment.kt

```diff
@@ -154,28 +151,18 @@ class BookshelfFragment : Fragment() {
     }
 
     private fun handleEvent(event: BookshelfViewModel.Event) {
-        val message =
-            when (event) {
-                is BookshelfViewModel.Event.OpenPublicationError -> {
-                    Timber.e(event.error.toDebugDescription())
-                    (event.error).toUserError().getUserMessage(requireContext())
-                }
+        when (event) {
+            is BookshelfViewModel.Event.OpenPublicationError -> {
+                activity?.let { event.error.toUserError().show(it) }
```

Comment:
Can `activity` be null? If so we can fail to deliver feedback. Fortunately, I doubt it.